kornia/color/adjust.py

```diff
@@ -225,6 +227,248 @@ def adjust_brightness(input: torch.Tensor,
 
 
+def _solarize(input: torch.Tensor, thresholds: Union[float, torch.Tensor] = 0.5) -> torch.Tensor:
+    r""" For each pixel in the image, select the pixel if the value is less than the threshold.
+        Otherwise, subtract 1.0 from the pixel.
+    Args:
+        input (torch.Tensor): image or batched images to solarize.
+        thresholds (float or torch.Tensor): solarize thresholds.
+            If int or one element tensor, input will be solarized across the whole batch.
+            If 1-d tensor, input will be solarized element-wise, len(thresholds) == len(input).
+    Returns:
+        torch.Tensor: Solarized images.
+    """
+    if not torch.is_tensor(input):
+        raise TypeError(f"Input type is not a torch.Tensor. Got {type(input)}")
+
+    if not isinstance(thresholds, (float, torch.Tensor,)):
+        raise TypeError(f"The factor should be either a float or torch.Tensor. "
+                        f"Got {type(thresholds)}")
+
+    if isinstance(thresholds, torch.Tensor) and len(thresholds.shape) != 0:
+        assert input.size(0) == len(thresholds) and len(thresholds.shape) == 1, \
+            f"threshholds must be a 1-d vector of shape ({input.size(0)},). Got {thresholds}"
+        # TODO: I am not happy about this line, but no easy to do batch-wise operation
+        thresholds = torch.stack([x.expand(*input.shape[1:]) for x in thresholds])
+
+    return torch.where(input < thresholds, input, 1.0 - input)
+
+
+def solarize(input: torch.Tensor, thresholds: Union[float, torch.Tensor] = 0.5,
+             additions: Optional[Union[float, torch.Tensor]] = 0.) -> torch.Tensor:
+    r""" For each pixel in the image less than threshold, we add 'addition' amount to it and then clip the
+        pixel value to be between 0 and 1.0. The value of 'addition' is between -0.5 and 0.5.
+    Args:
+        input (torch.Tensor): image or batched images to solarize.
+        thresholds (float or torch.Tensor): solarize thresholds.
+            If int or one element tensor, input will be solarized across the whole batch.
+            If 1-d tensor, input will be solarized element-wise, len(thresholds) == len(input).
+        additions (optional, float or torch.Tensor): between -0.5 and 0.5. Default None.
+            If None, no addition will be performed.
+            If int or one element tensor, same addition will be added across the whole batch.
+            If 1-d tensor, additions will be added element-wisely, len(additions) == len(input).
+    Returns:
+        torch.Tensor: Solarized images.
+    """
+    if not torch.is_tensor(input):
+        raise TypeError(f"Input type is not a torch.Tensor. Got {type(input)}")
+
+    if not isinstance(thresholds, (float, torch.Tensor,)):
+        raise TypeError(f"The factor should be either a float or torch.Tensor. "
+                        f"Got {type(thresholds)}")
+
+    if isinstance(thresholds, float):
+        thresholds = torch.tensor(thresholds)
+
+    if additions is not None:
```

Comment:
is None same as adding zeros ? actually default value is 0, right ?

Reply:
The outcome is the same. Using None could skip some calculations.